packages/hfs/src/linkFiles.ts

```diff
@@ -0,0 +1,94 @@
+import { switchMap } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { hfs } from '@capsulajs/capsula-api';
+import md5 from 'md5';
+
+let head = '';
+let body = '';
+const parseDep = (links, mod, dynamic = false) => {
+  if (links[mod] === undefined) {
+    throw new Error(`dependency ${mod} is not found`);
+  }
+  if (links[mod].processed) {
+    return;
+  }
+  links[mod].required = true;
+  if (dynamic) {
+    head += `const _${links[mod].root} = (obj) => new Promise(resolve => {import("./${links[mod].root}.js").then(mod => {resolve(mod.default(obj));})});\n`;
+  } else {
+    head += `import _${links[mod].root} from "./${links[mod].root}.js";\n`;
+  }
+
+  let dep = '';
+  for (const d in links[mod].dep) {
+    parseDep(links, d + '.js');
+    dep = dep
+      ? dep + ', ' + `${links[mod].dep[d]}: __${links[d + '.js'].root}`
+      : `${links[mod].dep[d]}: __${links[d + '.js'].root}`;
+  }
+  for (const d in links[mod].dyDep) {
+    parseDep(links, d + '.js', true);
+    dep = dep
+      ? dep + ', ' + `${links[mod].dyDep[d]}: __${links[d + '.js'].root}`
+      : `${links[mod].dyDep[d]}: __${links[d + '.js'].root}`;
+  }
+
+  if (dynamic) {
+    body += `const __${links[mod].root} = () => _${links[mod].root}({${dep}});\n`;
+  } else {
+    body += `const __${links[mod].root} = _${links[mod].root}({${dep}});\n`;
+  }
+
+  links[mod].processed = true;
+
+  return head + body;
+};
+const createLinksFile = (links, link) => {
+  head = '';
+  body = '';
+  return parseDep(links, link);
+};
+
+const linkFiles = switchMap((files: any[]) =>
+  Observable.create((obs) => {
```

Comment:
im not sure, isn't Observable.create deprecated?
```suggestion
 new  Observable((obs) => {
```